### PR TITLE
add wait-for-it.sh to start proxy after mysql instance available

### PR DESCRIPTION
**Pull request by cherrylzhao**
- add wait-for-it.sh
- optimize docker file

### CODE REVIEW

File: shardingsphere-integration-test/shardingsphere-proxy-docker-build/src/main/resources/tool/wait-for-it.sh [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+#   Use this script to test if a given TCP host/port are available
+
+WAITFORIT_cmdname=${0##*/}
+
+echoerr() { if [[ $WAITFORIT_QUIET -ne 1 ]]; then echo "$@" 1>&2; fi }
+
+usage()
+{
+    cat << USAGE >&2
```
Comment (kezhenxu94) [inline shardingsphere-integration-test/shardingsphere-proxy-docker-build/src/main/resources/tool/wait-for-it.sh:27]: I don't think you can simply add a license header when you copy a file from others, especially it's not licensed with AL2.0

https://github.com/vishnubob/wait-for-it/blob/master/wait-for-it.sh

Comment (kezhenxu94) [reply shardingsphere-integration-test/shardingsphere-proxy-docker-build/src/main/resources/tool/wait-for-it.sh:27]: FYI @tristaZero

Comment (tristaZero) [reply shardingsphere-integration-test/shardingsphere-proxy-docker-build/src/main/resources/tool/wait-for-it.sh:27]: @kezhenxu94 Hi, thanks for your kind remainder, and sorry for my lax checking. :)

Comment (cherrylzhao) [reply shardingsphere-integration-test/shardingsphere-proxy-docker-build/src/main/resources/tool/wait-for-it.sh:27]: ok, I'll revise it later, it's better wget it in docker file

Comment (cherrylzhao) [reply shardingsphere-integration-test/shardingsphere-proxy-docker-build/src/main/resources/tool/wait-for-it.sh:27]: fix it at #5765

**Pull request merged**